src/main/java/seedu/address/logic/commands/EditMeetingCommand.java

```diff
@@ -0,0 +1,280 @@
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Meeting> lastShownList = model.getFilteredMeetingList();
+        List<Contact> contactList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MEETING_DISPLAYED_INDEX);
+        }
+
+        Meeting meetingToEdit = lastShownList.get(index.getZeroBased());
+        Meeting editedMeeting = createEditedMeeting(meetingToEdit, editMeetingDescriptor, contactList);
+
+        if (!meetingToEdit.isSameMeeting(editedMeeting) && model.hasMeeting(editedMeeting)) {
+            throw new CommandException(MESSAGE_DUPLICATE_MEETING);
+        }
+
+        model.setMeeting(meetingToEdit, editedMeeting);
+        model.updateFilteredMeetingList(PREDICATE_SHOW_ALL_MEETINGS);
+        model.commitAddressBook();
+        return new CommandResult(String.format(MESSAGE_EDIT_MEETING_SUCCESS, editedMeeting));
+    }
+
+    /**
+     * Creates and returns a {@code Meeting} with the details of {@code meetingToEdit}
+     * edited with {@code editMeetingDescriptor}.
+     */
+    private static Meeting createEditedMeeting(Meeting meetingToEdit, EditMeetingDescriptor editMeetingDescriptor,
+                                               List<Contact> contactList) throws CommandException {
+        assert meetingToEdit != null;
+
+        MeetingName updatedMeetingName = editMeetingDescriptor.getMeetingName().orElse(meetingToEdit.getName());
+        MeetingDate updatedMeetingDate = editMeetingDescriptor.getMeetingDate().orElse(meetingToEdit.getDate());
+        StartTime updatedStartTime = editMeetingDescriptor.getStartTime().orElse(meetingToEdit.getStartTime());
+        EndTime updatedEndTime = editMeetingDescriptor.getEndTime().orElse(meetingToEdit.getEndTime());
+
+        if (updatedStartTime.isAfter(updatedEndTime)) {
+            throw new CommandException(MESSAGE_INVALID_TIME);
+        }
+
+        MeetingArchiveStatus archiveStatus = meetingToEdit.getArchiveStatus();
+        Set<Tag> updatedTags = editMeetingDescriptor.getTags().orElse(meetingToEdit.getTags());
+
+        final Set<Participant> updatedParticipants = new HashSet<>();
+
+        if (editMeetingDescriptor.getParticipantsIndex().isPresent()) {
+            for (Index targetIndex : editMeetingDescriptor.participantsIndex) {
+                if (targetIndex.getZeroBased() >= contactList.size()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+                }
+                Contact participatingContact = contactList.get(targetIndex.getZeroBased());
+                updatedParticipants.add(new Participant(participatingContact));
+            }
+        } else {
+            updatedParticipants.addAll(meetingToEdit.getParticipants());
+        }
```

Comment:
Just to note that there is some deep nesting here. I think we can work on it later on if there's no easy abstraction to fix it now.